vec.ecl

```diff
@@ -0,0 +1,116 @@
+/*##############################################################################
+## HPCC SYSTEMS software Copyright (C) 2020 HPCC Systems.  All rights reserved.
+############################################################################## */
+IMPORT $ as PBBlas;
+IMPORT Std.System.Thorlib;
+/**
+  *  This module handles different vector (column matrix) operations
+  */
+EXPORT vec := MODULE
+  SHARED node := Thorlib.node();
+  /**
+    *  Creation of a vector with given length and constant value.
+    *
+    *  @param N      Length of the vector in UNSIGNED.
+    *  @param def    Constant value of the vector in REAL.
+    *  @return       Matrix (vector) in DATASET(Layout_Cell) form.
+    *  @see          PBblas/Types.Layout_Cell
+    */
+  EXPORT From(UNSIGNED N,REAL8 def = 1.0) := FUNCTION
+    PerNode := ROUNDUP(N/CLUSTERSIZE);
+    // Create vector across all nodes
+    PBblas.Types.Layout_cell fillRow(UNSIGNED4 c) := TRANSFORM
+      SELF.wi_id := 1;
+      x := node * PerNode + c;
+      SELF.x := IF(x > N, SKIP, x);
+      SELF.y := 1;
+      SELF.v := def;
+    END;
+    // Now generate on each node
+    m := DATASET(PerNode, fillRow(COUNTER), LOCAL);
```

Comment:
> I actually want to use LOCAL rather than DISTRIBUTED on the DATASET.

Why?  To ensure all but the last node have PerNode records?  If so it would be worth commenting here.  This code might also not work if it is executed in a child query (although I can't think of a reason why it would....)

Reply:
@ghalliday I'll add a comment.  I assume LOCAL would give me the optimal performance since there would be no communication needed.

Reply:
Distributed also has no communication.  Each slave independently works out the range it needs to generate (from its node number and totoal # of nodes) and then generates the rowss

Reply:
@RogerDev